lib/features/family/features/reflect/bloc/grateful_cubit.dart

```diff
@@ -23,77 +24,99 @@ class GratefulCubit extends CommonCubit<GratefulUIModel, GratefulCustom> {
   final List<Organisation> _currentRecommendations = [];
 
   Future<void> init() async {
-    //TODO
     _initProfiles();
-    /*await _gratefulRecommendationsRepository
+
+    await _gratefulRecommendationsRepository
         .getGratefulRecommendationsForMultipleProfiles(_profiles);
-        
-    _currentRecommendations = await _gratefulRecommendationsRepository
-        .getGratefulRecommendations(_getCurrentProfile());*/
-    _emitData();
+
+    final _currentRecommendations = await _gratefulRecommendationsRepository
+        .getGratefulRecommendations(_getCurrentProfile());
+
+    _emitData(_currentRecommendations);
   }
 
   void _initProfiles() {
-    _profiles = _reflectAndShareRepository.getPlayers();
+    // Make copy of list to avoid mutation of original repository list
+    _profiles = List.from(_reflectAndShareRepository.getPlayers());
+
     _profiles
       ..sort((a, b) => a.isChild ? -1 : 1)
       ..removeWhere(
         (element) => !element.isChild || element.gratitude == null,
       ); //in the future we don't need to remove the adults
+
     if (_profiles.isEmpty) {
       _onEveryoneDonated();
     }
   }
 
   GameProfile _getCurrentProfile() => _profiles[_currentProfileIndex];
 
-  void _emitData() {
+  void _emitData(List<Organisation> currentRecommendations) {
```

Comment:
I would make currentRecommendations a field so the emitData does not need any params. Its a method that could be called in more cases than just retrieving recommendations so having to pass it all the time can become quite tedious.